### DISCUSSION TITLE
Parametr marketplaceId w /billing/billing-entries

### DISCUSSION THREAD

**Discussion by javaguru-pl** [allegro/allegro-api#7006] [Suggestions - rejected]
Widzę że w metodzie /billing/billing-entries pojawił się parametr marketplaceId, który przyjmuje domyślną wartość - marketplace bazowy. Czy oznacza to że nie można pobrać za jednym razem WSZYSTKICH billingów (z wszystkich marketplace-ów) tylko należy je pobierać po kolejnych marketplace-ach? To byłoby dodatkowe utrudnienie, gdyż zakładając że w przyszłości będzie wiele marketplace-ów - to będzie trzeba wykonywać każdorazowo requesty dla każdego z nich (nawet jeżeli sprzedawca nie jest na danym rynku aktywny).

Widzę że w metodzie /order/events - zastosowano dużo lepsze podejście - zwracane są wszystkie zdarzenia z wszystkich rynków (bez potrzeby określania marketplace). Czy można tak samo zrobić w /billing/billing-entries? Parametr marketplaceId może zostać, ale niech domyślnie przyjmuje wartość null - czyli zwraca koszty z wszystkich rynków.
Z góry dziękuję za pozytywne rozpatrzenie tej sugestii.

**Comment by MaciejFrackowiak** [comment]
@javaguru-pl Tak, zgadza się. Zwracamy domyślnie billing dla serwisu bazowego użytkownika. W billingu kluczowe jest saldo, a tylko zwracając je per marketplace możemy zachować jego ciągłość .

**Comment by javaguru-pl** [comment]
@MaciejFrackowiak ale wystarczy przecież w każdym zwracanym billing-entry zwracać dodatkowe pole marketplace-id (informujące którego marketplace dotyczy rekord) i nie będzie problemu z saldem. Po pobraniu billing-entries aplikacja już po swojej stronie podzieli je na osobne marketplace-y. Wydaje mi się że po stronie zespołu API brakuje czasem spojrzenia na API od strony użytkownika tego API. Moje podejście, które przedstawiłem (zwracanie rekordów z wszystkich rynków jednocześnie wraz z polem marketplace-id w każdym billing-entry) jest olbrzymim ułatwieniem dla użytkownika oraz zaoszczędza ilość wysyłanych requestów. Jak w przyszłości będzie np. z 10 marketplace-ów to strasznie uciążliwe będzie pobieranie billingów, dla użytkownika aktywnego np. na jednym marketplace - i tak będziemy musieli wykonać 10 requestów (z czego 9 będzie pustych). Proszę to poważnie rozważyć. Z góry dziękuję.

**Comment by MaciejFrackowiak** [comment]
@javaguru-pl Dziękuję za Twoje spojrzenie. Jeżeli zespół odpowiedzialny za usługę zdecyduje się na zmianę działania billingu, na pewno o tym poinformujemy.

**Comment by MartaNowaczyk** [comment]
Po weryfikacji przez odpowiedni zespół sugestię zwracania wszystkich billingów muszę odrzucić, bo jest to niemożliwe ze względu na podział billingu na subkonta, o które trzeba zapytać wprost.
